client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -361,9 +354,16 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
 
     private static Date parseDate(String value, DateTimeZone localTimeZone)
     {
-        long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
-        if (millis >= START_OF_MODERN_ERA_SECONDS * MILLISECONDS_PER_SECOND) {
-            return new Date(millis);
+        try {
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= START_OF_MODERN_ERA_SECONDS * MILLISECONDS_PER_SECOND) {
+                return new Date(millis);
+            }
+        }
+        catch (IllegalArgumentException ignored) {
+            // https://joda-time.sourceforge.net/faq.html#illegalinstant
```

Comment:
Can we somehow validate the exception. IllegalArgumentException looks generic.

Reply:
I've used `IllegalInstantException` instead